src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -1456,6 +1456,22 @@ internal override TypeSymbol SetUnknownNullabilityForReferenceTypes()
             return this.WithUnderlyingType(underlyingType);
         }
 
+        internal override bool TryMergeNullability(TypeSymbol other, out TypeSymbol merged)
+        {
+            var otherTuple = other as TupleTypeSymbol;
+            if ((object)otherTuple != null)
```

Comment:
>if [](start = 12, length = 2)

nit: Consider inlining as type pattern: `if (other is TupleTypeSymbol otherTuple) ...`